src/main/java/com/jcabi/log/VerboseProcess.java

```diff
@@ -473,6 +488,95 @@ public Void call() throws Exception {
             }
             return null;
         }
+
+        /**
+         * Logs supplied StringBuilder to supplied Logger and Writer.
+         * @param reader Reader to use
+         * @param writer Writer to use
+         * @throws IOException writer could throw this
+         * @throws ClosedByInterruptException thrown if interrupted
+         */
+        private void logFromReader(final BufferedReader reader,
+                final BufferedWriter writer) throws IOException,
+                ClosedByInterruptException {
+            StringBuilder builder = new StringBuilder();
+            String previous = NULL_STRING;
+            int lineCount = 0;
+            while (true) {
+                if (Thread.interrupted()) {
+                    Logger.debug(
+                        VerboseProcess.class,
+                        "explicitly interrupting read from buffer"
+                    );
+                    break;
+                }
+                if (previous != NULL_STRING) {
```

Comment:
@mkordas I used == because I actually want it to be the same instance in memory as NULL_STRING